wallet-new/test/unit/WalletUnitTest.hs

```diff
@@ -177,44 +183,106 @@ testPureWallet = do
     fullEmpty = walletBoot Full.walletEmpty . oursFromSet
 
 {-------------------------------------------------------------------------------
-  Passive wallet tests
+  Inductive wallet tests, compares DSL vs Cardano Wallet API results at each
+  inductive step.
 -------------------------------------------------------------------------------}
 
-testPassiveWallet  :: Spec
-testPassiveWallet = around bracketPassiveWallet $
-    describe "Passive wallet sanity checks" $ do
-      it "can be initialized" $ \w ->
-        Kernel.init w
+-- | GenInductive
+--
+-- Contains a Generated Inductive Wallet along with a DSL Wallet Builder
+-- (which embeds "ours" in the DSL world) and an ESK
+-- used to derive "ours" in the Cardano world.
+data GenInductive
+    = GenInductive (Inductive GivenHash Addr)
+                    -- ^ Generated Inductive Wallet
+                    (Transaction GivenHash Addr -> Wallet GivenHash Addr)
+                    -- ^ DSL Wallet Builder
+                    EncryptedSecretKey
+                    -- ^ ESK of selected poor actor
+
+testActiveWallet :: HasConfiguration => Spec
+testActiveWallet =
+    it "Test Active Wallet API - compare DSL and Cardano wallet state after each Inductive step" $
+      forAll genInductive $ \(GenInductive ind mkDSLWallet esk) ->
+                                (bracketActiveWallet $ \activeWallet -> do
+                                    checkEquivalent activeWallet esk mkDSLWallet ind)
 
--- | Initialize passive wallet in a manner suitable for the unit tests
-bracketPassiveWallet :: (Kernel.PassiveWallet -> IO a) -> IO a
-bracketPassiveWallet = Kernel.bracketPassiveWallet logMessage
   where
-   -- TODO: Decide what to do with logging
-    logMessage _sev _txt = return ()
+    genInductive :: Gen GenInductive
+    genInductive = do
+      -- generate a blockchain and capture the translation context
+      (fpc,transCtxt) <- runTranslateT $ do
+                                    transCtxt' <- ask
+                                    fpc' <- fromPreChain genValidBlockchain
+                                    return (fpc',transCtxt')
+
+      (poorIx, esk) <- choosePoorActor transCtxt
+
+      -- generate an Inductive Wallet for the blockchain, including ourAddrs
+      (InductiveWithOurs ourAddrs ind) <- genFromBlockchainWithOurs (ours' poorIx) fpc
+
+      return $ GenInductive ind (mkDSLWallet' transCtxt ourAddrs) esk
+
+    ours' :: ActorIx -> Addr -> Bool
+    ours' poorIx@(IxPoor _) addr = addrActorIx addr == poorIx
+    ours' _ _ = False
+
+    choosePoorActor :: TransCtxt -> Gen (ActorIx, EncryptedSecretKey)
+    choosePoorActor transCtxt' = do
+        let poorActors = Map.elems . actorsPoor . tcActors $ transCtxt'
+        poorIx <- choose (0, length poorActors -1)
+        let esk =  encKpEnc . poorKey $ poorActors !! poorIx
```

Comment:
If the intent is to pick a random element from `poorActors`, you can do `poorActor <- elements poorActors`